core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -734,7 +734,8 @@ class Partition(val topicPartition: TopicPartition,
     }
   }
 
-  def appendRecordsToLeader(records: MemoryRecords, isFromClient: Boolean, requiredAcks: Int = 0): LogAppendInfo = {
+  def appendRecordsToLeader(records: MemoryRecords, isFromClient: Boolean, apiVersion: ApiVersion = ApiVersion.latestVersion,
```

Comment:
I think it might be a little dangerous to rely on the caller to set the proper inter-broker version. I wonder if we could pass it to `Partition` on construction instead. Alternatively, we could dip into `replicaManager.config`, as we do in a couple other cases.

Reply:
Referencing `replicaManager.config` would be better, like `Partition#apply` does.